app/graphql/mutations/ingredients/create_ingredient.rb

```diff
@@ -0,0 +1,23 @@
+module Mutations
+  module Ingredients
+    class CreateIngredient < Mutations::BaseMutation
+      argument :params, Types::Input::Ingredients::IngredientInputType, required: true
+
+      field :ingredient, Types::IngredientType, null: false
+
+      def resolve(params:)
+        recipe = recipe_exists?(params[:recipe_id])
+        if recipe
+          ingredient = recipe.ingredients.create(params.to_h)
+          { ingredient: ingredient }
+        else
+          GraphQL::ExecutionError.new("No record of Recipe with ID #{params[:recipe_id]}")
+        end
+      end
+
+      def recipe_exists?(id)
+        Recipe.find(id) rescue false
+      end
```

Comment:
Noticed this also exists in another file. Maybe we can pull out to module to DRY up later.

Reply:
Good thinking. I'll keep an eye out for other little pieces of code that could be put in a module in the future